fastlane/lib/fastlane/actions/set_info_plist_value.rb

```diff
@@ -29,13 +36,66 @@ def self.run(params)
             File.write(path, new_plist)
           end
 
+          return plist if params[:map]
           return params[:value]
         rescue => ex
           UI.error(ex)
           UI.user_error!("Unable to set value to plist file at '#{path}'")
         end
       end
 
+      # Adapted from https://stackoverflow.com/a/30225093/158525
+      # rubocop:disable Style/CaseEquality
+      # rubocop:disable Style/MultilineTernaryOperator
+      # rubocop:disable Style/NestedTernaryOperator
+      def self.deep_merge(a, b)
+        merger = proc do |key, v1, v2|
+          Hash === v1 && Hash === v2 ?
+                 v1.merge(v2, &merger) : Array === v1 && Array === v2 ?
+                   array_deep_merge(v1, v2) : [:undefined, nil, :nil].include?(v2) ? v1 : v2
+        end
+        a.merge(b, &merger)
+      end
+      # rubocop:enable Style/CaseEquality
+      # rubocop:enable Style/MultilineTernaryOperator
+      # rubocop:enable Style/NestedTernaryOperator
+
+      # Adapted from https://github.com/danielsdeleo/deep_merge/blob/86f15a04bc6cee3723125de6165a6b43b55f1cde/lib/deep_merge/core.rb#L174-L186
+      def self.array_deep_merge(source, dest)
+        source_all_hashes = source.all? { |i| i.kind_of?(Hash) }
+        dest_all_hashes = dest.all? { |i| i.kind_of?(Hash) }
+        if source_all_hashes && dest_all_hashes
+          # merge hashes in lists
+          list = []
+          dest.each_index do |i|
+            list[i] = deep_merge(source[i] || {}, dest[i])
+          end
+          list += source[dest.count..-1] if source.count > dest.count
+          dest = list
+        else
+          dest = source | dest
+        end
+        dest
+      end
+
+      def self.deep_stringify(obj)
+        if obj.kind_of?(Hash)
+          stringified_hash ||= {}
+          obj.each do |k, v|
+            stringified_hash[k.to_s] = deep_stringify(v)
+          end
+          return stringified_hash
+        end
+        if obj.kind_of?(Array)
+          stringified_array ||= []
+          obj.each do |v|
+            stringified_array << deep_stringify(v)
+          end
+          return stringified_array
+        end
+        obj
+      end
```

Comment:
Thanks for not including the extra dependency, can we move these into their own class or maybe monkey patches of Array and/or Hash?

Reply:
Not too familiar with Ruby, so unsure 🤔 
But sounds like a good idea.

Reply:
If we move it into it's own class we can reuse it on slack and frameit, and hopefully we won't break anything 👍

Reply:
@ohayon any suggestion on what this util class would be called and where to place it? so we can reuse it

Reply:
Some monkey patches for `String` are at `fastlane\fastlane_core\lib\fastlane_core\core_ext\string.rb` for example. Would probably make sense to move these over there or in a similar file that is included in a similar way.